src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -0,0 +1,253 @@
+          '@_selected': true,
+        });
+      }
+      if (_.has(answer, 'unselectedFeedback') && _.get(answer, 'unselectedFeedback', '').trim() !== '') {
+        feedback.push({
+          '#text': _.get(answer, 'unselectedFeedback'),
+          '@_selected': false,
+        });
+      }
+      if (_.has(answer, 'feedback') && _.get(answer, 'feedback', '').trim() !== '') {
+        feedback.push({
+          '#text': _.get(answer, 'feedback'),
+        });
+      }
+      if (feedback.length) {
+        singleAnswer[`${option}hint`] = feedback;
+      }
+      singleAnswer = {
+        '#text': answer.title,
+        '@_correct': answer.correct,
+        ...singleAnswer,
+      };
+      choice.push(singleAnswer);
+    });
+    widget = { [option]: choice };
+    if (_.has(this.problemState, 'groupFeedbackList')) {
+      compoundhint = this.addGroupFeedbackList();
+      widget = {
+        ...widget,
+        compoundhint,
+      };
+    }
+    return widget;
+  }
+
+  addGroupFeedbackList() {
+    const compoundhint = [];
+    const { groupFeedbackList } = this.problemState;
+    groupFeedbackList.forEach((element) => {
+      compoundhint.push({
+        '#text': element.feedback,
+        '@_value': element.answers.join(' '),
+      });
+    });
+    return compoundhint;
+  }
+
+  addQuestion() {
+    const { question } = this.problemState;
+    const questionObject = this.parser.parse(question);
+    return questionObject;
+  }
+
+  buildMultiSelectProblem(problemType, widget, option) {
+    const question = this.addQuestion();
+    const widgetObject = this.addMultiSelectAnswers(option);
+    const demandhint = this.addHints();
+    const problemObject = {
+      problem: {
+        [problemType]: {
+          ...question,
+          [widget]: widgetObject,
+        },
+        ...demandhint,
+      },
+    };
+    return this.builder.build(problemObject);
+  }
+
+  buildTextInput() {
+    const question = this.addQuestion();
+    const demandhint = this.addHints();
+    const answerObject = this.buildTextInputAnswersFeedback();
+    const problemObject = {
+      problem: {
+        [OLXProblemTypes.TEXTINPUT]: {
+          ...question,
+          ...answerObject,
+        },
+        ...demandhint,
+      },
+    };
+    return this.builder.build(problemObject);
+  }
+
+  buildTextInputAnswersFeedback() {
+    const { answers } = this.problemState;
+    let answerObject = {};
+    const additionAnswers = [];
+    const wrongAnswers = [];
+    let firstCorrectAnswer = false;
+    answers.forEach((answer) => {
+      const correcthint = this.getAnswerHints(answer);
+      if (answer.correct && firstCorrectAnswer) {
+        additionAnswers.push({
+          '@_answer': answer.title,
+          ...correcthint,
+        });
+      } else if (answer.correct && !firstCorrectAnswer) {
+        firstCorrectAnswer = true;
+        answerObject = {
+          '@_answer': answer.title,
+          ...correcthint,
+        };
```

Comment:
The variable name `firstCorrectAnswer` and its usage is very confusing.
Either rename it something like `firstCorrectAnswerParsed` or do:
```suggestion
    let firstCorrectAnswer = true;
    answers.forEach((answer) => {
      const correcthint = this.getAnswerHints(answer);
      if (answer.correct && firstCorrectAnswer) {
        firstCorrectAnswer = false;
        answerObject = {
          '@_answer': answer.title,
          ...correcthint,
        };
      } else if (answer.correct && !firstCorrectAnswer) {
        additionAnswers.push({
          '@_answer': answer.title,
          ...correcthint,
        });
```